internals/controllers/migrationhierarchy_controller.go

```diff
@@ -145,6 +146,17 @@ func (r *MigrationHierarchyReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 		// resync for the parent source sns
 		r.addSnsToSnsEvent(sourceSnsParentName, sourceSnsParentNamespace)
+
+		// the root sns for resync every sns childrens displayname
+		snsToSync, err := utils.NewObjectContext(ctx, log, r.Client, client.ObjectKey{Name: ns.GetName(), Namespace: utils.GetNamespaceParent(ns.Object)}, &danav1.Subnamespace{})
+		if err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
I don't understand the comment. Also, don't we already have this object elsewhere in the function?